### ISSUE TITLE
[Question] Does a vertically integrated Harvester > cluster > Rancher automatic deployment make sense?

### ISSUE THREAD

**Issue by ndizazzo** [harvester/harvester#2004]
I learned about Harvester when looking for alternatives to Proxmox and just today discovered that it's possible to create a HA K3S cluster by adding Harvester as a cloud provider in an existing Rancher installation and deploying a new cluster.

**This means I could:** 
* Set up Rancher on an existing cluster
* Install Harvester and register it as a cloud provider in Rancher
* Ask Rancher to create _new_ cluster on VMs from within Rancher for deploying workloads to. 
  
In the end, though, you would end up with:
* 1 Harvester installation
* 1 Rancher installation
* 2 K3S clusters

**But doesn't it make more sense to do this from the other direction starting with Harvester? IE:** 
* Install Harvester
* Deploy VMs that automatically configure HA K3S nodes (Bonus: join bare-metal etc/control plane/worker to the cluster)
* Automatically install Rancher 
* Automatically link Harvester as a cloud provider to Rancher

It seems to me like the smallest building block to using both products together is starting with Harvester + VMs _then_ layering in a cluster + Rancher. One could do all of this by hand, but a single integrated pane of glass is way more fun :) 

I think there would be a lot of questions to answer like what OS runs the K3S cluster, how does a K3S version get installed and started, what version of Rancher gets installed (lots of dropdowns?)... but in the end think it'd be great because we'd end up with 1 Harvester install, 1 Rancher install, and 1 K3S cluster along with the ability to mix in bare-metal machines to the newly deployed cluster.

Does this make much sense as a feature?

**Comment by fadjar340** [comment]
+1

**Comment by radioflyer28** [comment]
I miss the automatic Rancher install/link too for multi-cluster management. If I recall right, in v0.3 one could elect to simply enable this from the dashboard ui. I was surprised to find a separate Rancher install was needed when moving to v1.0.0.

**Comment by tlehman** [comment]
To clarify, this is the automated install setup you are imagining?

![2004](https://user-images.githubusercontent.com/261552/167036683-7e40de52-e323-4895-85b4-22f7f192c97c.jpg)

**Comment by radioflyer28** [comment]
@tlehman I believe that's what @ndizazzo and I are saying. However, I recently discovered a hidden UI that I wasn't aware of: https://10.0.0.10/dashboard/c/local/explorer
It at least offers some ability to deploy workloads directly via Rancher without some of Rancher's other bells and whistles, like multi-cluster management. This is generally what I was looking for and appears to be what I used in v0.3.

Although, it sounds like @ndizazzo wants a full Rancher install.
